backend/typescript/middlewares/validators/camperValidators.ts

```diff
@@ -0,0 +1,83 @@
+import { Request, Response, NextFunction } from "express";
+import { getApiValidationError, validatePrimitive } from "./util";
+
+/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
+/* eslint-disable-next-line import/prefer-default-export */
+export const updateCamperDtoValidator = async (
+  req: Request,
+  res: Response,
+  next: NextFunction,
+) => {
+  if (req.body.firstName && !validatePrimitive(req.body.firstName, "string")) {
+    return res.status(400).send(getApiValidationError("firstName", "string"));
+  }
+  if (req.body.lastName && !validatePrimitive(req.body.lastName, "string")) {
+    return res.status(400).send(getApiValidationError("lastName", "string"));
+  }
+  if (req.body.age && !validatePrimitive(req.body.age, "string")) {
+    return res.status(400).send(getApiValidationError("age", "string"));
+  }
+  if (
+    req.body.parentName &&
+    !validatePrimitive(req.body.parentName, "string")
+  ) {
+    return res.status(400).send(getApiValidationError("parentName", "string"));
+  }
+  if (
+    req.body.contactEmail &&
+    !validatePrimitive(req.body.contactEmail, "string")
+  ) {
+    return res
+      .status(400)
+      .send(getApiValidationError("contactEmail", "string"));
+  }
+  if (
+    req.body.contactNumber &&
+    !validatePrimitive(req.body.contactNumber, "string")
+  ) {
+    return res
+      .status(400)
+      .send(getApiValidationError("contactNumber", "string"));
+  }
+  if (req.body.camp && !validatePrimitive(req.body.camp, "string")) {
```

Comment:
we should also validate that the camp that is passed in has the same parent camp as the old camp, since we only allow changing between sessions of a camp!